app/src/main/java/pub/devrel/easypermissions/sample/SingleCallbackFragment.java

```diff
@@ -0,0 +1,68 @@
+package pub.devrel.easypermissions.sample;
+
+import android.Manifest;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.Fragment;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import java.util.List;
+
+import pub.devrel.easypermissions.AfterPermissionGranted;
+import pub.devrel.easypermissions.EasyPermissions;
+
+/**
+ *  Created in {@link R.layout#activity_main}
+ */
+public class SingleCallbackFragment extends Fragment implements EasyPermissions.PermissionCallback {
+
+    private static final String TAG = "SingleCallbackFragment";
+    private static final int RC_SMS_PERM = 122;
+
+    @Override
+    public View onCreateView (LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        super.onCreateView(inflater, container, savedInstanceState);
+
+        // Create view
+        View v =  inflater.inflate(R.layout.fragment_main, container);
+
+        // Button click listener
+        v.findViewById(R.id.button_sms).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                smsTask();
+            }
+        });
+
+        return v;
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+
+        // EasyPermissions handles the request result.
+        EasyPermissions.onRequestPermissionsResult(requestCode, permissions, grantResults, this);
+    }
+
+    @AfterPermissionGranted(RC_SMS_PERM)
+    private void smsTask() {
+        if (EasyPermissions.hasPermissions(getContext(), Manifest.permission.READ_SMS)) {
+            // Have permission, do the thing!
+            Toast.makeText(getActivity(), "TODO: SMS things", Toast.LENGTH_LONG).show();
+        } else {
+            // Request one permission
+            EasyPermissions.requestPermissions(this, getString(R.string.rationale_sms),
+                    RC_SMS_PERM, Manifest.permission.READ_SMS);
+        }
+    }
+
+    @Override
+    public void onPermissionsResults(int requestCode, List<String> grantedPerms, List<String> deniedPerms) {
+        Log.d(TAG, "onPermissionsResults:" + requestCode + ":granted:" + grantedPerms.size() + ":denied:"+ deniedPerms.size());
```

Comment:
Sorry about the long delay.  Here's my major hesitation on this PR:  is this much more more convenient than just  doing more logic in `onRequestPermissionsResult`?

Something like this:

```java
public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
    // Forward to system and easypermissions
    super.onRequestPermissionsResult(requestCode, permissions, grantResults);
    EasyPermissions.onRequestPermissionsResult(requestCode, permissions, grantResults, this);

    List<String> granted = new ArrayList<>;
    List<String> denied = new ArrayList<>;

    for (String perm : permissions) {
      if (EasyPermissions.hasPermission(this, perm) {
          granted.add(perm);
      } else {
          denied.add(perm);
      }
    }
```

While obviously your solution is less boilerplate, there's nothing fundamentally hard about doing what I did above, and adding the single callback mode in a backwards compatible way results in a lot more internal logic in the library.

Is there another use case I am not thinking of?  I really do appreciate how thorough this PR is, but I worry that the cost exceeds the benefit.